File: components/json/src/test/scala/io/eels/component/json/JsonSourceTest.scala
package io.eels.component.json

import com.sksamuel.scalax.io.IO
import io.eels.{FrameSchema, Row}
import org.apache.hadoop.fs.Path
import org.scalatest.{Matchers, WordSpec}

class JsonSourceTest extends WordSpec with Matchers {

  import scala.concurrent.ExecutionContext.Implicits.global

  "JsonSource" should {
    "read multiple json docs from a file" in {
      val expectedSchema = FrameSchema("name", "location")
      JsonSource(new Path(IO.fileFromResource("/test.json").getAbsolutePath)).toSet shouldBe
        Set(
          Row(expectedSchema, "sammy", "aylesbury"),
          Row(expectedSchema, "ant", "greece")
        )
    }
  }
}

File: eel-core/src/test/kotlin/io/eels/component/json/JsonSourceTest.kt
package io.eels.component.json

import io.eels.Row
import io.eels.schema.Field
import io.eels.schema.Schema
import io.kotlintest.specs.WordSpec
import org.apache.hadoop.fs.Path

class JsonSourceTest : WordSpec() {
  init {
    "JsonSource" should {
      "read multiple json docs from a file" {
        val schema = Schema(Field("name"), Field("location"))
        JsonSource(Path(javaClass.getResource("/test.json").file)).toFrame(1).toSet() shouldBe
            setOf(
                Row(schema, "sammy", "aylesbury"),
                Row(schema, "ant", "greece")
            )
      }
    }
  }
}